### DISCUSSION TITLE
Share artifacts, army, exchange castle/town

### DISCUSSION THREAD

**Discussion by denierlexiese** [ihhub/fheroes2#8997] [Ideas]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

I love the gift option in multiplayer. If it could be extended to pass artifacts to team mate somehow, whether horse to horse, or new gui that would be amazing. Additionally, if there was a way to exchange army characters too with team mate that would be amazing.  Also if there was a way my team mate could walk into my empty castle/town and I could give that to them that would be great too.

### Describe the possible solution

I love the gift option in multiplayer. If it could be extended to pass artifacts to team mate somehow, whether horse to horse, or new gui that would be amazing. Additionally, if there was a way to exchange army characters too with team mate that would be amazing.  Also if there was a way my team mate could walk into my empty castle/town and I could give that to them that would be great too.

### Additional info

None. Please let me know if the request doesn't make sense, or is already achievable through some game option.

**Comment by Cheerfulbull** [comment]
Another idea is to share the Obelisk puzzle reveal, the way map discoveries are shared.

**Comment by ihhub** [comment]
Moving this topic to the discussion section. Such ideas require not only logical explanation and heavy UI changes.
